Assignment 1/src/main/java/com/cezaram28/Assignment1/repository/jdbc/JdbcVoteRepository.java

```diff
@@ -0,0 +1,105 @@
+package com.cezaram28.Assignment1.repository.jdbc;
+
+import com.cezaram28.Assignment1.entity.Answer;
+import com.cezaram28.Assignment1.entity.Question;
+import com.cezaram28.Assignment1.entity.User;
+import com.cezaram28.Assignment1.entity.Vote;
+import com.cezaram28.Assignment1.repository.VoteRepository;
+import lombok.RequiredArgsConstructor;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.jdbc.core.simple.SimpleJdbcInsert;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+@RequiredArgsConstructor
+public class JdbcVoteRepository implements VoteRepository {
+
+    private final JdbcTemplate template;
+
+    @Override
+    public Vote save(Vote vote) {
+        if (vote.getId() == null) {
+            vote.setId(insert(vote));
+        } else {
+            update(vote);
+        }
+        return vote;
+    }
+
+    @Override
+    public void remove(Vote vote) {
+        template.update("DELETE FROM vote WHERE id=?", vote.getId());
+    }
+
+    void updateVote(Vote v) {
+        JdbcUserRepository jdbcUserRepository = new JdbcUserRepository(template);
+        Optional<User> u = jdbcUserRepository.findById(v.getUser().getId());
+        JdbcQuestionRepository jdbcQuestionRepository = new JdbcQuestionRepository(template);
+        Optional<Question> q = jdbcQuestionRepository.findById(v.getQuestion().getId());
+        JdbcAnswerRepository jdbcAnswerRepository = new JdbcAnswerRepository(template);
+        Optional<Answer> a = jdbcAnswerRepository.findById(v.getAnswer().getId());
+        if(u.isPresent())
```

Comment:
Generally it is a good practice to use curly brackets for your if, else, for, while statements even if they contain a single instruction.